File: signal-detector/src/foam.ts
import * as foam from '@foam-ai/node-opentelemetry';
import { FOAM_API_KEY, IS_PRODUCTION } from './config';
import { ExpressInstrumentation } from '@opentelemetry/instrumentation-express';
import type { InstrumentationBase } from '@opentelemetry/instrumentation';

foam.init({
  serviceName: 'signal-detector',
  isProduction: IS_PRODUCTION,
  apiKey: FOAM_API_KEY,
  additionalInstrumentations: [
    new ExpressInstrumentation({}) as unknown as InstrumentationBase,
  ],
});

File: ai-agent-service/src/foam.py
import foam
from src.config import FOAM_API_KEY, IS_PRODUCTION

foam.init(
    service_name="ai-agent-service",
    is_production=IS_PRODUCTION,
    api_key=FOAM_API_KEY,
    traces_sample_rate=1.0 if not IS_PRODUCTION else 0.3,
)